website/client/store/actions/shops.js

```diff
@@ -112,12 +112,13 @@ export function purchaseMysterySet (store, params) {
 }
 
 export function purchaseHourglassItem (store, params) {
+  const quantity = params.quantity || 1;
   const user = store.state.user.data;
-  let opResult = hourglassPurchaseOp(user, {params});
+  let opResult = hourglassPurchaseOp(user, {params, quantity});
 
   return {
     result: opResult,
-    httpCall: axios.post(`/api/v4/user/purchase-hourglass/${params.type}/${params.key}`),
+    httpCall: axios.post(`/api/v4/user/purchase-hourglass/${params.type}/${params.key}`, {quantity}),
```

Comment:
if quantity would be also a `/{params.quantity}`-param, then it wouldn't have to be in body

Reply:
I think when you add a param it would still accept the route (without the newly added one)